### PR TITLE
yaml_decode: add example and associated tests

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: yaml_decode/example/example.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'package:json_annotation/json_annotation.dart';
```
Comment (jakemac53) [inline yaml_decode/example/example.dart]: This example doesn't directly use any apis from this package?

Comment (kevmoo) [reply yaml_decode/example/example.dart]: The annotation, my friend. 😉

Comment (jakemac53) [reply yaml_decode/example/example.dart]: I get that JsonSerializable uses it, but that doesn't serve as an example for using this package?

Comment (kevmoo) [reply yaml_decode/example/example.dart]: Sorry. Doing crawl walk run. Going to add in a CLI here that parses yaml.

Comment (kevmoo) [reply yaml_decode/example/example.dart]: ...and associated tests, etc

Comment (jakemac53) [reply yaml_decode/example/example.dart]: Is it worth having this at all for now though?

Comment (kevmoo) [reply yaml_decode/example/example.dart]: Trying to keep the PRs smaller. I'm going to build on this next – I promise. 😄

**Pull request merged**